src/main/java/com/uid2/admin/audit/QLDBAuditMiddleware.java

```diff
@@ -0,0 +1,57 @@
+package com.uid2.admin.audit;
+
+import com.uid2.admin.Constants;
+import com.uid2.admin.vertx.service.IService;
+import com.uid2.shared.auth.IAuthorizable;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+
+public class QLDBAuditMiddleware implements AuditMiddleware{
+    private final AuditWriter auditWriter;
+
+    protected QLDBAuditMiddleware(AuditWriter writer){
+        this.auditWriter = writer;
+    }
+
+    @Override
+    public Handler<RoutingContext> handle(AuditHandler<RoutingContext> handler) {
+        InnerAuditHandler auditHandler = new InnerAuditHandler(handler, auditWriter);
+        return auditHandler::writeLog;
+    }
+
+    private static class InnerAuditHandler{
+        private final AuditHandler<RoutingContext> innerHandler;
+        private final AuditWriter auditWriter;
+        private InnerAuditHandler(AuditHandler<RoutingContext> handler, AuditWriter auditWriter) {
+            this.innerHandler = handler;
+            this.auditWriter = auditWriter;
+        }
+
+        public void writeLog(RoutingContext rc){
+            List<OperationModel> modelList = innerHandler.handle(rc);
+            if(modelList != null && rc.statusCode() < 300){
```

Comment:
Why check status code? Shouldn't non-null operation  means something needs to be logged?

Reply:
modelList == null means there is nothing to log (this can occur if the endpoint fails due to malformed input, or auth forbids access). Status code check can be removed; this was a relic of older code